### ISSUE TITLE
Update/Add Clang 15.0.0+

### ISSUE THREAD

**Issue by IGR2014** [actions/runner-images#6533]
### Tool name

Clang

### Tool license

Apache License v2.0 with LLVM Exceptions

### Add or update?

- [X] Add
- [X] Update

### Desired version

15.0.0+

### Approximate size

676 MB

### Brief description of tool

C-like languages use the [Clang](http://clang.llvm.org/) frontend. This component compiles C, C++, Objective-C, and Objective-C++ code into LLVM bitcode -- and from there into object files, using LLVM.

### URL for tool's homepage

http://clang.llvm.org/

### Provide a basic test case to validate the tool's functionality.

```bash
clang++ --version
```

### Platforms where you need the tool

- [X] Azure DevOps
- [X] GitHub Actions

### Runner images where you need the tool

- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [X] Ubuntu 22.04
- [ ] macOS 10.15
- [ ] macOS 11
- [ ] macOS 12
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Can this tool be installed during the build?

```bash
apt-get install -y clang-15 lldb-15 lld-15 clang-format-15 clang-tidy-15
```

### Tool installation time in runtime

Approx. 5 minutes

### Are you willing to submit a PR?

Yes if possible

**Comment by al-cheb** [comment]
Hey @IGR2014.
clang-15 is not available in Ubuntu 22.04 repos - https://packages.ubuntu.com/search?keywords=clang-15

**Comment by IGR2014** [comment]
Oh, so only 22.04 official packages for 22.04 ?
Thanks, got it...
